src/core/Akka.Remote/EndpointManager.cs

```diff
@@ -340,44 +373,62 @@ protected override SupervisorStrategy SupervisorStrategy()
                 var directive = Directive.Stop;
 
                 ex.Match()
-                    .With<InvalidAddressAssociationException>(ia =>
+                    .With<InvalidAssociation>(ia =>
                     {
-                        log.Warning("Tried to associate with unreachable remote address [{0}]. Address is now gated for {1} ms, all messages to this address will be delivered to dead letters. Reason: [{2}]",
-                                 ia.RemoteAddress, settings.RetryGateClosedFor.TotalMilliseconds, ia.Message);
-                        endpoints.MarkAsFailed(Sender, Deadline.Now + settings.RetryGateClosedFor);
-                        AddressTerminatedTopic.Get(Context.System).Publish(new AddressTerminated(ia.RemoteAddress));
+                        KeepQuarantinedOr(ia.RemoteAddress, () =>
```

Comment:
When an `InvalidAssociation` occurs, instead of immediately marking the remote system as failed and publishing an `AddressTerminated` message (which causes all watched actors on that remote system to be `Terminated`,)  we now go and check to see if this system has a refuseUID in place.

* If we do have a refuseUID, then we **quarantine** the remote system.
* If we don't have a refuseUID, then the remote system is unreachable and **fail* it.